### PR TITLE
[feature](executor) Use the heartbeat BE info to set pipeline variables. 

### PR DESCRIPTION
**Pull request by Mryange**
# Proposed changes

Use the CPU core count returned by BE heartbeat information to set the parallel_fragment_exec_instance_num variable to half of the minimum value in all BE.
you can use 'show backends' to get it.

image

## Problem summary

Some designs may require optimization, and a new variable may be introduced in the future, initialized to -1. If the user makes changes, the content provided by the user is used, otherwise, the content provided by BE is used.

Because of the fuzzy reason, there is a test here that has fuzzy turned off. https://github.com/apache/doris/pull/19212 

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/qe/BackendInfo.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        Info.clear();
+        mincorSize = 9999;
+    }
+
+    public static void addBeInfo(long beId, long coreSize) {
+        Info.put(beId, new BackendInfo(coreSize));
+        mincorSize = Math.min(mincorSize, coreSize);
+    }
+
+    public static long getMincorSize() {
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/qe/BackendInfo.java:53]:
```suggestion
    public static long getMinNumCores() {
```

File: gensrc/thrift/HeartbeatService.thrift [hunk 2]
```diff
@@ -43,6 +43,7 @@ struct TBackendInfo {
     5: optional string version
     6: optional i64 be_start_time
     7: optional string be_node_role
+    8: optional i64 core_size
```
Comment (Gabriel39) [inline gensrc/thrift/HeartbeatService.thrift:46]:
```suggestion
    8: optional i64 num_cores
```

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.96 seconds
 stream load tsv:          426 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230426140938_clickbench_pr_135924.html

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by morningman — CHANGES_REQUESTED** [review]
Better not using heartbeat logic to do that.
We need to keep heartbeat logic as simple as we can.

And there are other way to sync info between BE and FE.
For example, there is a disk report logic that will report disk info of BE to FE at fix interval.
You can modify that logic to implement your need.

**Pull request closed without merging**